programs/models.py

```diff
@@ -367,6 +368,54 @@ def __str__(self):
         return self.name.text
 
 
+class WarningMessageManager(models.Manager):
+    translated_fields = ("message",)
+
+    def new_warning(self, calculator, external_name=None):
+        translations = {}
+        for field in self.translated_fields:
+            translations[field] = Translation.objects.add_translation(f"warning.{calculator}_temporary_key-{field}")
+
+        if external_name is None:
+            external_name = calculator
+
+        # try to set the external_name to the name
+        external_name_exists = self.filter(external_name=external_name).count() > 0
+
+        warning = self.create(
+            external_name=external_name if not external_name_exists else None,
+            calculator=calculator,
+            **translations,
+        )
+
+        for [field, translation] in translations.items():
+            translation.label = f"navigator.{calculator}_{warning.id}-{field}"
+            translation.save()
+
+        return warning
+
+
+class WarningMessage(models.Model):
+    programs = models.ManyToManyField(Program, related_name="warning_messages", blank=True)
+    external_name = models.CharField(max_length=120, blank=True, null=True, unique=True)
+    calculator = models.CharField(max_length=120, blank=False, null=False)
```

Comment:
while adding a new warning message, what can be the value for the `calculator` field?

Reply:
Currently is can be: "_show", "_dont_show", or "_tax_unit".

This is defined in: programs/programs/warnings/__init__.py

This also lets us create custom calculators. For example, if we wanted to display a message on a program if they have children that are younger than a certain age, we could make a warning calculator for that.